scipy/stats/tests/test_multivariate.py

```diff
@@ -1860,18 +1860,25 @@ def test_det_and_ortho(self):
         dets = np.array([[np.linalg.det(x) for x in xx] for xx in xs])
         assert_allclose(np.fabs(dets), np.ones(dets.shape), rtol=1e-13)
 
-        # Test that we get both positive and negative determinants
-        # Check that we have at least one and less than 10 negative dets in a sample of 10. The rest are positive by the previous test.
-        # Test each dimension separately
-        assert_array_less([0]*10, [np.nonzero(d < 0)[0].shape[0] for d in dets])
-        assert_array_less([np.nonzero(d < 0)[0].shape[0] for d in dets], [10]*10)
-
         # Test that these are orthogonal matrices
         for xx in xs:
             for x in xx:
                 assert_array_almost_equal(np.dot(x, x.T),
                                           np.eye(x.shape[0]))
 
+    @pytest.mark.parametrize("dim", [2, 5, 10, 20])
+    def test_det_distribution_gh18272(self, dim):
+        # Test that positive and negative determinants are equally likely.
+        rng = np.random.default_rng(6796248956179332344)
+        dist = ortho_group(dim=dim)
+        rvs = dist.rvs(size=5000, random_state=rng)
+        dets = np.linalg.det(rvs)
```

Comment:
```suggestion
        dets = scipy.linalg.det(rvs)
```
@ilayn it should just be a drop-in replacement, right? Luckily, `scipy.linalg` is imported at the top, so feel free to commit if this looks right.

Reply:
Yes, fingers crossed

Reply:
IIRC it's automatically done in CI before tests run. Anyway, I committed and tests passed.

Reply:
Phew! 😅